### DISCUSSION TITLE
How to Expose 2 graphql endpoints using netflix dgs library with spring boot?

### DISCUSSION THREAD

**Discussion by mkarthik-git** [Netflix/dgs-framework#1276] [Q&A]
Is there a way to expose 2 graphql endpoints using spring boot netflix dgs library? As alternate, I have seen the post of using ServletRegistrationBean where we can define a GraphQLHttpServlet as a bean, and configure the context path to be /graphql, but this one uses kickstart-library. Looking for similar pattern with using netflix dgs library with spring.

**Comment by srinivasankavitha** [comment]
Hi - We don't currently support exposing 2 graphql endpoints.

**Comment by Navodar22** [comment]
I'd also vote for possibility to expose multiple endpoints. 
I'd like to have a 1 endpoint **/public/graphql** hidden behind for example saml and second endpoints **/internal/graphql** hidden behind different security.

**Comment by srinivasankavitha** [comment]
Thanks for the input. We will add this for upcoming features. However,
since this is not a high priority for us internally, the feature may take
some time to deliver. We do welcome contributions in the meantime.

**Comment by KambleShilpa** [comment]
@srinivasankavitha Hello! I want to expose two different schemas for two different endpoints. I wanted to know if the said feature is implemented?

**Comment by srinivasankavitha** [comment]
HI @KambleShilpa - we don't have the ability to expose 2 different schemas in the current framework.

**Reply by renatoabreu11** [reply]
Hey @srinivasankavitha,
Is there any roadmap for this specific feature request or any news on the topic? 
I find the feature can be particular useful for exposing an external endpoint and an internal endpoint, allowing a better separation of data and concerns. 
I am trying to implement this customization in one of my services by playing around with the webmvc configuration, however without success so far.

**Reply by srinivasankavitha** [reply]
We don't have it prioritized on the roadmap yet since there isn't quite a need internally for it. You are welcome to open a PR for this feature as well since we do accept contributions to the framework.
